utils/tx.ts

```diff
@@ -1168,6 +1166,67 @@ export async function updateFeeLockMetadata(
   );
 }
 
+export async function multiSwapBuyMarket(
+  user: User,
+  tokenIds: BN[],
+  buyAmount: BN,
+  maxAmountIn: BN = MAX_BALANCE,
+) {
+  const api = await getApi();
+  const swapPoolList: BN[] = [];
+  let i: number = 0;
+  let liqId: BN;
+  const tokenIdsLength = tokenIds.length;
+  const firstToken = tokenIds[0];
+  const lastToken = tokenIds[tokenIdsLength - 1];
+  while (i < tokenIdsLength - 1) {
+    liqId = await getLiquidityAssetId(tokenIds[i], tokenIds[i + 1]);
+    swapPoolList.push(liqId);
+    i++;
+  }
```

Comment:
Ive seen this code duplicated more than 2 times, I think is worth it to extract it to a fn

Reply:
Done